src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -90,62 +105,119 @@ public ShooterSubsystem(ShooterHoodSubsystem shooterHood) {
+    lastTime = Timer.getFPGATimestamp();
   }
 
-  /**
-   * Passes a preset velocity to the SparkMAX PIDF controller and lets it manage
-   * the NEO's velocity. Intended to be called when a button is pressed.
-   */
   public void shoot() {
-    // extend hood
     // fast shooter wheel
     // run gate if both of "" are good
+    // NetworkTableInstance.getDefault().getTable("limelight").getEntry("ledMode").setNumber(3);
+
+    if (enablePID) {
+      shooterPIDRight.setReference(MoPrefs.getShooterPIDSetpoint(), ControlType.kVelocity);
+    } else {
+      leader_shooterMAXRight.set(MoPrefs.getShooterFlywheelSetpoint());
+    }
+
+    final boolean shooterHoodReady = shooterHood.hasReliableZero() && shooterHood.getFullyDeployed();
+    final boolean shooterWheelReady = MoPrefs.getShooterFlywheelSetpoint()
+        - leader_shooterMAXRight.getEncoder().getVelocity() < MoPrefs.getShooterFlywheelTolerance();
+    if (shooterHoodReady && shooterWheelReady) {
+      shooterGate.set(MoPrefs.getShooterGateSetpoint());
+    } else {
+      shooterGate.set(0);
+    }
+
+  }
+
+  public void shootAuto() {
+    // NetworkTableInstance.getDefault().getTable("limelight").getEntry("ledMode").setNumber(3);
     shooterHood.deployHood();
-    shooterPIDRight.setReference(MoPrefs.getShooterFlywheelSetpoint(), ControlType.kVelocity);
-    if (shooterHood.getFullyDeployed()
-        && MoPrefs.getShooterFlywheelSetpoint() - shooterMAXRight.getEncoder().getVelocity() < 0.1) {
+
+    leader_shooterMAXRight.set(MoPrefs.getShooterFlywheelSetpoint());
+
+    if (shooterHood.hasReliableZero() && shooterHood.getFullyDeployed() && MoPrefs.getShooterFlywheelSetpoint()
+        - leader_shooterMAXRight.getEncoder().getVelocity() < MoPrefs.getShooterFlywheelTolerance()) {
       shooterGate.set(MoPrefs.getShooterGateSetpoint());
     } else {
       shooterGate.set(0);
     }
   }
 
   public void idle() {
-    // stow hood
     // stop gate
     // slow shooter wheel
-    shooterHood.stowHood();
-    shooterPIDRight.setReference(MoPrefs.getShooterFlywheelIdle(), ControlType.kVelocity);
     shooterGate.stopMotor();
+    // NetworkTableInstance.getDefault().getTable("limelight").getEntry("ledMode").setNumber(1);
+
+    if (maintainFlywheelAtIdle) {
+      if (enablePID) {
+        shooterPIDRight.setReference(MoPrefs.getShooterFlywheelIdle(), ControlType.kVelocity);
+      } else {
+        leader_shooterMAXRight.set(MoPrefs.getShooterFlywheelIdle());
+      }
+    } else {
+      if (enablePID) {
+        shooterPIDRight.setReference(0, ControlType.kVelocity);
+      } else {
+        leader_shooterMAXRight.stopMotor();
+      }
+    }
   }
 
   public void purge() {
     // stow hood
     // reverse gate
     // reverse shooter wheel
-    shooterHood.stowHood();
-    shooterPIDRight.setReference(-MoPrefs.getShooterFlywheelIdle(), ControlType.kVelocity);
-    shooterGate.set(-MoPrefs.getShooterGateSetpoint());
+    if (enablePID) {
```

Comment:
No need for PID here.  Always use dumb speed.

No need for -1 * either, just use -
-MoPrefs.getShooterFlywheelIdle()